src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,196 @@
+        Pⱼ₋₁, (1/j) * ((2j-1) * X .* Pⱼ₋₁ - (j-1) * Pⱼ₋₂)
+      I += ((2j+1)/2) * (mean(Pⱼ₋₁))^2
+    end
+  end
+  return I
+end
+
+function optimaxis(transform, Z)
+  _, q = size(Z)
+  Iₘₐₓ, iₘₐₓ = 0, 0
+  for i = 1:q
+    eᵢ = [1. * (j==i) for j = 1:q]
+    I = pindex(transform, Z, eᵢ)
+    if I > Iₘₐₓ
+      Iₘₐₓ, iₘₐₓ = I, i
+    end
+  end
+  [1. * (iₘₐₓ==i) for i = 1:q]
+end
+
+function optimdiag(transform, Z, ᾱ)
+  l  = length(ᾱ)
+  I₀ = pindex(transform, Z, ᾱ)
+  for i in 1:l
+    eᵢ = [1. * (j==i) for j = 1:l]
+    f₊ = pindex(transform, Z, (1/√(2+2ᾱ[i])) * (ᾱ + eᵢ))
+    f₋ = pindex(transform, Z, (1/√(2-2ᾱ[i])) * (ᾱ - eᵢ)) * (ᾱ[i] != 1)
+    f = (f₊ > f₋) * f₊ + (f₊ ≤ f₋) * f₋ 
+    s = (f₊ > f₋) * 1 + (f₊ ≤ f₋) * (-1)
+    if f > pindex(transform, Z, ᾱ)
+      ᾱ = (1/√(2+2s*ᾱ[i])) * (ᾱ + s*eᵢ)
+    end
+  end
+  pindex(transform, Z, ᾱ) ≥ I₀ ? ᾱ : optimaxis(transform, Z)
+end
+
+function optimgen(transform, Z, α₀)
+  f(α) = -pindex(transform, Z, α)
+  op = optimize(f, α₀)
+  Optim.minimizer(op)
+end
+
+function optim(transform, Z)
+  α_axis = optimaxis(transform, Z)
+  α_diag = optimdiag(transform, Z, α_axis)
+  optimgen(transform, Z, α_diag)  
+end
+
+# transforms the cdf of a convex combination of r.v. to a standard normal cdf
+function structure_removal(transform, Z, α)
+  l = length(α)
+  U = qr(hcat(α, rand(l, l-1)))
+  if norm(diag(U.R)) < transform.tol
+    return print("Error: the random matrix is singular. Try again")
+  end
+  T = Z * U.Q
+  table = Tables.table(T)
+  newtable, _ = apply(Quantile(), table)
+  T[:,1] = newtable[:,1]
+  return T * U.Q'
+end
+
+# gives the array that will be used to stop the loop of structure 
+# removal by computing the projection index on samples of
+# standard multivariate Gaussian distribution
+function gauss_pmatrix(transform, N, q)
+  I = rand(transform.n, q)
+  d = MvNormal(zeros(q), Diagonal(ones(q)))
+  Q = qr(rand(q, q)).Q
+  for i = 1:transform.n
+    S = rand(d, N)'
+    for j = 1:q
+      I[i,j] = pindex(transform, S, Q[:,j])
+    end
+  end
+  I, Q
+end
+
+#Computes the projection index of each loop dataset (with possibly 
+#modified structure) along q random directions and compares with the 
+#array given by gaussian_projection_matrix.
+function stop_criteria(transform, Z, I, Q)
+  k, q = size(I)
+  switch = true
+  for i = 1:q
+    b = pindex(transform, Z, Q[:,i])
+    switch *= quantile(I[1:k-1,i], transform.perc) > b
+  end
+  return !switch
+end
+
+#Generates the loop for the structure removal process
+function apply(transform::ProjectionPursuit, table) 
+
+  names = Tables.columnnames(table)
+  newtable, _ = apply(seq, table)
+  Z = Tables.matrix(newtable)
+  N,q = size(Z)
+  I, Q = gauss_pmatrix(transform, N, q) 
+  count = 0
+  
+  while stop_criteria(transform, Z, I, Q)
+    α = optim(transform, Z)
+    Z = structure_removal(transform, Z, α)
+    count += 1
+    if count ≥ transform.maxiter
+      return "Maximum iteration ($count) reached!"
+    end
```

Comment:
```suggestion
    if count ≥ transform.maxiter
      @warn "Maximum iteration ($count) reached!"
      break
    end
```
The `apply` function must always return a table and a cache.
If the maximum number of iterations is reached, a warn will be displayed to the user and the current result will be returned.

Reply:
Thanks for the suggestion, I'll do it